src/main/java/com/redhat/pantheon/sling/PantheonInitializer.java

```diff
@@ -1,15 +1,97 @@
 package com.redhat.pantheon.sling;
 
+import org.apache.jackrabbit.api.JackrabbitSession;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlList;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlManager;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlPolicy;
+import org.apache.jackrabbit.api.security.principal.PrincipalManager;
+import org.apache.jackrabbit.api.security.user.User;
 import org.apache.sling.jcr.api.SlingRepository;
 import org.apache.sling.jcr.api.SlingRepositoryInitializer;
+import org.osgi.service.component.annotations.Component;
+
+import javax.jcr.PropertyType;
+import javax.jcr.RepositoryException;
+import javax.jcr.Value;
+import javax.jcr.ValueFactory;
+import javax.jcr.security.AccessControlPolicy;
+import javax.jcr.security.Privilege;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /**
  * Created by ben on 3/7/19.
  */
+@Component(service = SlingRepositoryInitializer.class)
 public class PantheonInitializer implements SlingRepositoryInitializer {
 
     @Override
     public void processRepository(SlingRepository slingRepository) throws Exception {
-        //May be a good place to autocreate users at some point in the future
+        JackrabbitSession s = (JackrabbitSession) slingRepository.loginAdministrative(null);
+        try {
+            User admin = (User) s.getUserManager().getAuthorizable("admin");
+            admin.changePassword("ccsadmin"); // FIXME - hardcoding admin passwords is a Bad Thing
+
+            // http://jackrabbit.apache.org/api/2.16/org/apache/jackrabbit/core/security/authorization/GlobPattern.html
+            assignPermissionToPrincipal(s, "anonymous", "/content/modules", "/*/cachedContent*", Privilege.JCR_MODIFY_PROPERTIES); // No idea why the trailing * is necessary but it doesn't work without it
+            assignPermissionToPrincipal(s, "demo", "/content/modules", null, Privilege.JCR_WRITE, Privilege.JCR_NODE_TYPE_MANAGEMENT);
```

Comment:
I think this might be the ticket: https://sling.apache.org/documentation/bundles/repository-initialization.html

Reply:
Yeah, I had also found that page - that's how I was originally clued in to the SlingRepositoryInitializer class. I guess you're referring to their repoinit language.

I have mixed feelings about it. On one hand, it could potentially simplify our code, which is always a good thing. On the other hand, it's one more layer of abstraction that will make troubleshooting that much more difficult - and in particular, this appears to be a custom language, so our troubleshooting efforts will ultimately boil down to "guess and check." I'm never a fan of such constructs.

But, you could also argue that if we're able to use the scripts, that makes our project more future-proof against API changes.

So I'm on the fence about it. I see arguments both ways. If you feel strongly, I can look into figuring it out.

Reply:
No, I don't necessarily feel strongly about it. I'll leave that call to you.